src/Microsoft.Identity.Client/IPublicClientApplication.cs

```diff
@@ -287,5 +289,152 @@ public partial interface IPublicClientApplication : IClientApplicationBase
+        /// token which is granted upon successful login by the user based on the device code information</description></item>
+        /// </list>
+        /// See https://aka.ms/msal-device-code-flow.
+        /// </summary>
+        /// <param name="scopes">Scopes requested to access a protected API</param>
+        /// <param name="extraQueryParameters">This parameter will be appended as is to the query string in the HTTP authentication request to the authority. 
+        /// This is expected to be a string of segments of the form <c>key=value</c> separated by an ampersand character.
+        /// The parameter can be null.</param>
+        /// <param name="deviceCodeResultCallback">Callback containing information to show the user about how to authenticate and enter the device code.</param>
+        /// <returns>Authentication result containing a token for the requested scopes and for the user who has authenticated on another device with the code</returns>
+
+        Task<AuthenticationResult> AcquireTokenWithDeviceCodeAsync(
+            IEnumerable<string> scopes,
+            string extraQueryParameters,
+            Func<DeviceCodeResult, Task> deviceCodeResultCallback);
+
+        /// <summary>
+        /// Acquires a security token on a device without a Web browser, by letting the user authenticate on 
+        /// another device, with possiblity of cancelling the token acquisition before it times out. This is done in two steps:
+        /// <list type="bullet">
+        /// <item><description>the method first acquires a device code from the authority and returns it to the caller via
+        /// the <paramref name="deviceCodeResultCallback"/>. This callback takes care of interacting with the user
+        /// to direct them to authenticate (to a specific URL, with a code)</description></item>
+        /// <item><description>The method then proceeds to poll for the security
+        /// token which is granted upon successful login by the user based on the device code information. This step is cancelable</description></item>
+        /// </list>
+        /// See https://aka.ms/msal-device-code-flow.
+        /// </summary>
+        /// <param name="scopes">Scopes requested to access a protected API</param>
+        /// <param name="deviceCodeResultCallback">The callback containing information to show the user about how to authenticate and enter the device code.</param>
+        /// <param name="cancellationToken">A CancellationToken which can be triggered to cancel the operation in progress.</param>
+        /// <returns>Authentication result containing a token for the requested scopes and for the user who has authenticated on another device with the code</returns>
+        Task<AuthenticationResult> AcquireTokenWithDeviceCodeAsync(
+            IEnumerable<string> scopes,
+            Func<DeviceCodeResult, Task> deviceCodeResultCallback,
+            CancellationToken cancellationToken);
```

Comment:
>CancellationToken [](start = 12, length = 17)

Aside, @MarkZuber: is there an issue tracking getting CancellationToken generally into the library?

Reply:
No, there isn't but there should be.